src/main/java/uk/gov/hmcts/reform/civil/service/CoreCaseDataService.java

```diff
@@ -159,10 +160,19 @@ public LocalDate getAgreedDeadlineResponseDate(Long caseId, String authorization
         return null;
     }
 
-    public SearchResult getCasesUptoMaxsize(String authorization) {
-        String query = new SearchSourceBuilder().size(RETURNED_NUMBER_OF_CASES)
-            .query(QueryBuilders.matchAllQuery()).toString();
+    public SearchResult getCCDDataBasedOnIndex(String authorization, int startIndex) {
+        String query = new SearchSourceBuilder()
+            .query(QueryBuilders.matchAllQuery())
+            .sort("data.submittedDate", SortOrder.DESC)
+            .from(startIndex)
+            .size(RETURNED_NUMBER_OF_CASES).toString();
 
         return coreCaseDataApi.searchCases(authorization, authTokenGenerator.generate(), CASE_TYPE, query);
     }
+
+    public int getTotalCCDCasesCount(String authorization) {
```

Comment:
Does the above query not return total cases? Why are we doing 2 queries?

Reply:
The above query return ,  but it fetches the records based on the sorting and  indexing and total count, but this query only returns the total records count, not records which are very required  for  total pages  before fetching the records, so written separately